snapshot-labs/snapshot
# Space settings: use alias for vote and proposal

Fixes Issue #2844 

**Background**

Currently voting and creating new proposals requires a signature each time. The goal of this PR is to add space setting enabling using the alias for both actions (one setting enables both).

**Related PRs**
- [snapshot.js](https://github.com/snapshot-labs/snapshot.js/pull/746)
- [snapshot-hub](https://github.com/snapshot-labs/snapshot-hub/pull/492)

**Changes in this PR:**
- [x] update `interfaces` and `queries` with new `aliased` field for `spaces.filters` and `spaces.voting`
- [x] add inputs and locale for aliased proposal and voting
- [x] check alias and pass alias data in `useClient` composable if space settings are enabled

**Screenshots & Rec**

Voting setting (currently in voting section, doesn't mention proposals):
![image](https://user-images.githubusercontent.com/39556343/197531675-92eaf8a2-e3e4-427f-a8d7-6b72d859ec66.png)

Voting without alias:
![unaliased](https://user-images.githubusercontent.com/39556343/197532590-31257c9a-6c93-4384-854a-088836d4c355.gif)

Aliased proposal & vote:
![aliased](https://user-images.githubusercontent.com/39556343/197532621-e786d057-ed4a-4b77-8a39-59ba6a878d1a.gif)

**How to test and review this PR?**
1. Go to [fabien.eth settings](http://localhost:8080/#/fabien.eth/settings) and enable aliased voting (requires merging two PRs mentioned above)
2. Create a new proposal for the space - if you have connected your wallet, there shouldn't be a request for signature
3. New proposal should be created
4. Vote on proposal - shouldn't get a request for signature and vote should go through :heavy_check_mark:
